native/src/modules/endpoint/sagas/loadCityContent.js

```diff
@@ -40,20 +126,15 @@ export default function* loadCityContent(
   yield call(dataContainer.storeLastUsage, newCity, criterion.peeking())
 
   if (!criterion.peeking()) {
-    const appSettings = new AppSettings()
-    const settings: SettingsType = yield call(appSettings.loadSettings)
-
-    if (settings.allowPushNotifications) {
-      yield spawn(NotificationsManager.subscribeNews, newCity, newLanguage, buildConfig().featureFlags)
-    }
-
-    yield call(appSettings.setSelectedCity, newCity)
+    yield call(selectCity, newCity, newLanguage)
   }
 
   const lastUpdate: Moment | null = yield call(dataContainer.getLastUpdate, newCity, newLanguage)
-  console.debug('Last city content update on ', lastUpdate ? lastUpdate.toISOString() : 'never')
+  console.debug('Last previous city content update: ', lastUpdate ? lastUpdate.toISOString() : 'never')
 
-  const shouldUpdate = criterion.shouldUpdate(lastUpdate)
+  const netInfo = yield call(NetInfo.fetch)
+  // Only try to update if there is an internet connection
+  const shouldUpdate = criterion.shouldUpdate(lastUpdate) && netInfo.isInternetReachable
```

Comment:
Not sure if this makes sense. That way a force refresh does not fail if there is no internet connection won't fail and no warning will be shown.

Reply:
This is indeed problematic. Can we check here whether it was a force refresh and then yield an error action?

Reply:
I think I'll just revert the change. An error action is already yielded if there is no internet connection.